templates/component.ejs

```diff
@@ -14,7 +14,8 @@ const styles = StyleSheet.create({
 
 type Props = {};
 
-<% if (props.isConnected) { %>class <%= props.name %> extends Component<Props> {<% } else { %>export default class <%= props.name %> extends Component<Props> {<% } %>
+<% if (props.isConnected) { %><% if (props.isPure) { %>class <%= props.name %> extends PureComponent<Props> {<% } else { %>class <% props.name %> extends Component<Props> {<% } %><% } else { %><% if (props.isPure) { %>export default class <%= props.name %> extends PureComponent<Props> {<% } else { %>export default class <% props.name %> extends Component<Props> {<% } %><% } %>
```

Comment:
You could define a variable in ejs in here : 
`<% var typeOfComponent = props.isPure ? 'PureComponent' : 'Component' %>`
and use that everywhere

Reply:
@adrienthiery At last, I got around to it, thanks!